packages/scheduler/src/SchedulerMinHeap.js

```diff
@@ -58,15 +56,16 @@ function siftUp(heap, node, i) {
 function siftDown(heap, node, i) {
   let index = i;
   const length = heap.length;
-  while (index < length) {
+  const halfLength = length >>> 1;
+  while (index < halfLength) {
     const leftIndex = (index + 1) * 2 - 1;
     const left = heap[leftIndex];
     const rightIndex = leftIndex + 1;
     const right = heap[rightIndex];
```

Comment:
@Tom910 Isn't the algorithm going beyond the array length here potentially? Otherwise we wouldn't need to check if `rightIndex < length` later, right?

Or does v8 only actually go through the prototype chain once we actually use the value e.g. when doing `right !== undefined`?

Or is de-opting here fine? You mentioned benchmarks in your PR description but I couldn't find where they are located. Could you share them for (future) testing?